packages/rpc-methods/src/restricted/dialog.ts

```diff
@@ -0,0 +1,298 @@
+import {
+  PermissionSpecificationBuilder,
+  PermissionType,
+  RestrictedMethodOptions,
+  ValidPermissionSpecification,
+} from '@metamask/controllers';
+import { assertExhaustive } from '@metamask/snap-utils';
+import { hasProperty, isObject, NonEmptyArray } from '@metamask/utils';
+import { ethErrors } from 'eth-rpc-errors';
+
+const methodName = 'snap_dialog';
+
+export enum DialogType {
+  alert = 'alert',
+  confirmation = 'confirmation',
+  prompt = 'prompt',
+}
+
+export type AlertFields = {
+  /**
+   * The alert title, no greater than 40 characters long.
+   */
+  title: string;
+
+  /**
+   * A description, displayed with the title, no greater than 140 characters
+   * long.
+   */
+  description?: string;
+
+  /**
+   * Free-from text content, no greater than 1800 characters long.
+   */
+  textAreaContent?: string;
+};
+
+export type ConfirmationFields = {
+  /**
+   * A question describing what the user is confirming, no greater than 40
+   * characters long.
+   */
+  title: string;
+
+  /**
+   * A description, displayed with the question, no greater than 140 characters
+   * long.
+   */
+  description?: string;
+
+  /**
+   * Free-from text content, no greater than 1800 characters long.
+   */
+  textAreaContent?: string;
+};
+
+export type PromptFields = {
+  /**
+   * The prompt title, no greater than 40 characters long.
+   */
+  title: string;
+
+  /**
+   * A description, displayed with the prompt, no greater than 140 characters
+   * long.
+   */
+  description?: string;
+};
+
+export type DialogFields = AlertFields | ConfirmationFields | PromptFields;
+
+type ShowAlert = (snapId: string, fields: AlertFields) => Promise<null>;
```

Comment:
Do we need 3 hooks instead of one that just includes the `type`? 🤔 

It could be handled by one hook that has the switch statement in the extension, but not sure if we prefer handling it here or in the extension

Reply:
We can either have three hooks or on with a parameter that switches on the type. I'm agnostic.